dotnet/src/Agents/Abstractions/Definition/AgentToolDefinition.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.SemanticKernel.Agents;
+
+/// <summary>
+/// The options for defining a tool.
+/// </summary>
+public class AgentToolDefinition
+{
+    /// <summary>
+    /// The type of the tool.
+    /// </summary>
+    /// <remarks>
+    /// Used to identify which type of tool is being used e.g., code interpreter, openapi, ...
+    /// </remarks>
+    public string? Type
+    {
+        get => this._type;
+        set
+        {
+            Verify.NotNull(value);
+            this._type = value;
+        }
+    }
+
+    /// <summary>
+    /// The name of the tool.
+    /// </summary>
+    public string? Name
+    {
+        get => this._name;
+        set
+        {
+            Verify.NotNull(value);
+            this._name = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets the configuration for the tool.
+    /// </summary>
+    /// <remarks>
+    /// Used to store tool specific configuration e.g., files associated with the tool, etc.
+    /// </remarks>
+    [JsonExtensionData]
+    public IDictionary<string, object>? Configuration
+    {
+        get => this._configuration;
```

Comment:
Since both are definitions I would keep `Configuration` or `Metadata` for the ExtensionData.

AgentDefinition.Metadata -> Configuration
OR 
AgentToolDefinition.Configuration -> Metadata

```suggestion
        get => this._configuration ??= new Dictionary<string, object>();
```